lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1157,6 +1157,19 @@ void TypeAnalysis::visitOperation(Operation *op,
     return;
   }
 
+  if (isa<AtenFftFftOp>(op)) {
```

Comment:
@ramiro050 if I remove this code from RefineTypes, it stops being able to infer the fft type. Not sure if I'm missing something.

Reply:
This probably means the dtype function is not being simplified correctly. Let me take a look

Reply:
You need to add handling here: https://github.com/llvm/torch-mlir/blob/64f9a0e978d530a2c00d2134e6a01c41cb034b9a/lib/Dialect/Torch/Utils/Utils.cpp#L77

Reply:
Ah perfect yeah that was the issue! Thanks for looking into it!

One final question. I removed my lit tests from `refine-types-ops.mlir` since I removed the code from `RefineTypes.cpp`. So where can I place lit tests now with this new flow? `reify-dtype-calculations.mlir`?

Reply:
I would just add one test in `simplify-dtype-calculations.mlir` showing that if the `dtypes` region returns a complex dtype, the right `dtype` gets assigned to the op. Something like this test:

https://github.com/llvm/torch-mlir/blob/64f9a0e978d530a2c00d2134e6a01c41cb034b9a/test/Dialect/Torch/simplify-dtype-calculations.mlir#L14